lib/producer.js

```diff
@@ -43,12 +43,20 @@ function isMessageAttributeValid(messageAttribute) {
 }
 
 function entryFromObject(message) {
-  if (!message.id || !message.body) {
-    throw new Error('Object messages must have \'id\' and \'body\' props');
+  if (!message.body) {
+    throw new Error('Object messages must have \'body\' prop');
+  }
+    
+  if (!message.groupId && !message.deduplicationId && !message.id) {
+    throw new Error('Object messages must have \'id\' prop');
+  }
+    
+  if (message.deduplicationId && !message.groupId) {
+    throw new Error('FIFO Queue messages must have \'groupId\' prop');
   }
 
   var entry = {
-    Id: message.id,
+    Id: message.id || message.body,
```

Comment:
This change is not tested, and I'm not sure it's necessary? This would always assign an ID to any message, but this whole PR is about _not_ needing an id?